### PR TITLE
[fix] use same where condition in user analytics as in pings for mau

### PR DESCRIPTION
**Pull request by kopancek**
Previously we used a different condition to calculate the MAU and other stats in user analytics page. Now we use the same condition as we use for pings. We also calculate everything using UTC now, so both should give the same value.

## Test plan

Tested locally and also with data from Tinder. The new query for MAU seems to be returning the correct data for us, I manually cross-referenced the old query vs the new. Apart from the timezone difference, the new query is more precise as in it filters the sourcegraph-operator users and non-user events in the same way as we do for the pings.

### CODE REVIEW

File: internal/adminanalytics/users.go [hunk 1]
```diff
@@ -40,11 +40,12 @@ const (
 	WITH user_days_used AS (
         SELECT
             CASE WHEN user_id = 0 THEN anonymous_user_id ELSE CAST(user_id AS TEXT) END AS user_id,
-            COUNT(DISTINCT DATE(timestamp)) AS days_used
+            COUNT(DISTINCT DATE(TIMEZONE('UTC', timestamp))) AS days_used
```
Comment (erzhtor) [inline internal/adminanalytics/users.go:43]: Curious, why do we need this? I thought timestamps are stored in UTC anyway, aren't they?

File: internal/adminanalytics/utils.go [hunk 2]
```diff
@@ -126,50 +69,26 @@ SELECT
...
-	}
-
-	return conds, nil
+func getDefaultConds() []*sqlf.Query {
+	return database.BuildCommonUsageConds(&database.CommonUsageOptions{
+		ExcludeSystemUsers:          true,
+		ExcludeNonActiveUsers:       true,
+		ExcludeSourcegraphAdmins:    true,
+		ExcludeSourcegraphOperators: true,
+	}, []*sqlf.Query{})
```
Comment (erzhtor) [inline internal/adminanalytics/utils.go:82]: I like the way you simplified this! 💪

File: internal/adminanalytics/utils.go [hunk 3]
```diff
@@ -126,50 +69,26 @@ SELECT
...
 }
 
-func makeEventLogsQueries(ctx context.Context, db database.DB, cache bool, dateRange string, grouping string, events []string, conditions ...*sqlf.Query) (*sqlf.Query, *sqlf.Query, error) {
+func makeEventLogsQueries(dateRange string, grouping string, events []string, conditions ...*sqlf.Query) (*sqlf.Query, *sqlf.Query, error) {
```
Comment (erzhtor) [inline internal/adminanalytics/utils.go:85]: Not for this PR I think, but would be great to have some tests for this func as it is being used in almost all the in-product analytics. P.S: I regret that we didn't add tests for this previously.

**Review by erzhtor — APPROVED** [review]
Thanks for these changes, looks cleaner! Besides minor questions/comments looks good to me!

File: internal/adminanalytics/users.go [hunk 4]
```diff
@@ -163,12 +161,7 @@ func (f *Users) MonthlyActiveUsers(ctx context.Context) ([]*MonthlyActiveUsersRo
 	prevMonth := now.AddDate(0, -2, 0) // going back 2 months
```
Comment (thenamankumar) [inline internal/adminanalytics/users.go:161]: The previous month timeframe calculations can be troubling too.

**Review by thenamankumar — APPROVED** [review]
These changes give the exact numbers as the pings?

File: internal/adminanalytics/users.go (hunk 1 above)
Comment (kopancek) [reply internal/adminanalytics/users.go:43]: No, the default timezone in the database is whatever the system it is running on default is. So in my case it's using EU timezone.

File: internal/adminanalytics/utils.go (hunk 3 above)
Comment (kopancek) [reply internal/adminanalytics/utils.go:85]: Agreed. Adding a note for myself for a followup.

**Pull request merged**